src/lib/services/workflow-service.ts

```diff
@@ -206,17 +207,28 @@ export const fetchAllArchivedWorkflows = async (
 export async function fetchWorkflow(
   parameters: GetWorkflowExecutionRequest,
   request = fetch,
-): Promise<WorkflowExecution> {
+): Promise<{
+  workflow: WorkflowExecution | undefined;
```

Comment:
Noticed that workflow or error doesn't actually show up as possibly `undefined` ⬇️  Wondering if there's a better way to type this so it does...

![Screenshot 2024-01-24 at 11 34 35 AM](https://github.com/temporalio/ui/assets/15069288/625ec39e-efff-47be-974d-cbdd3108518f)

Also wondering if we shouldn't just return the object without the `undefined` property (e.g. `{ error: e }`.

Reply:
Good call